starter-code/scripts/article.js

```diff
@@ -50,13 +50,23 @@ Article.fetchAll = function() {
     // When rawData is already in localStorage,
     // we can load it with the .loadAll function above,
     // and then render the index page (using the proper method on the articleView object).
-    Article.loadAll(); //TODO: What do we pass in to loadAll()?
+    Article.loadAll(JSON.parse(localStorage.getItem('rawData'))); //TODO: What do we pass in to loadAll()?
     //TODO: What method do we call to render the index page?
+    articleView.initIndexPage();
   } else {
     // TODO: When we don't already have the rawData,
     // we need to retrieve the JSON file from the server with AJAX (which jQuery method is best for this?),
     // cache it in localStorage so we can skip the server call next time,
     // then load all the data into Article.all with the .loadAll function above,
-    // and then render the index page.
+    // and then render the index page.'
+    $.getJSON('data/hackerIpsum.json', function(data) {
+      localStorage.setItem('rawData', JSON.stringify(data));
+      Article.loadAll(data);
+    });
+    articleView.initIndexPage();
```

Comment:
`articleView.initIndexPage();` actually needs to be inside of the `$.getJSON` function.
This is because the `$.getJSON` is asynchronous, but you want to wait until the data is successfully retrieved before calling the initializing the index page.